### PR TITLE
Gremble0 41 api post add created at and updated at to posts

### LINKED ISSUE #41: API post: add created at and updated at to posts
Schema file should include created at and updatet at in the schema file. The posts from the GET route {domain}/posts should include created at and updated at. 

Routes potentially effected:

GET route {domain}/posts
POST route {domain}/posts (the date should automatically be initiated)
PUT route {domain}/posts/id 
GET route {domain}/posts/id

Example:
```
"posts": [
      {
        "id": 0,
        "content": "string",
        "createdAt": "date time",
        "updatedAt": "date time",
        "author": {
          "id": 0,
          "cohort_id": 0,
          "role": "string",
          "firstName": "string",
          "lastName": "string",
          "bio": "string",
          "githubUsername": "string",
          "profileImageUrl": "string"
        }
      }
    ]
```

### PR DESCRIPTION
**Pull request by Panfi98**

### CODE REVIEW

**Review by gremble0 — DISMISSED** [review]

**Review by thomaafl — DISMISSED** [review]

File: prisma/seed.js [hunk 1]
```diff
@@ -123,7 +123,9 @@ async function createPost(userId, content) {
   const post = await prisma.post.create({
     data: {
       userId,
-      content
+      content,
+      createdAt: new Date(),
+      updatedAt: new Date()
```
Comment (dogezen) [inline prisma/seed.js:278]: no need to add createdAt and updatedAt - these are inserted automatically by Prisma

File: src/controllers/post.js [hunk 2]
```diff
@@ -30,6 +30,8 @@ export const create = async (req, res) => {
     const newPost = await dbClient.post.create({
       data: {
         content,
+        createdAt: new Date(),
+        updatedAt: new Date(),
```
Comment (dogezen) [inline src/controllers/post.js]: same here, prisma will take care of thes

File: prisma/schema.prisma [hunk 3]
```diff
@@ -56,6 +56,8 @@ model Post {
   id       Int       @id @default(autoincrement())
   content  String
   userId   Int
+  createdAt DateTime    @default(now())
+  updatedAt DateTime    @updatedAt
```
Comment (dogezen) [inline prisma/schema.prisma]: have you tested running this migration on a table that didn't ahve these 2 columns but had existing rows? i think the migration will probably break on the updatedAt.

I think you need to create 2 migrations:

MODEL for first migration:

```
createdAt DateTime    @default(now())
  updatedAt DateTime    @default(now()) // set an initial value for all updatedAt for each existing row
```

generate the migration file for the above model, THEN

change model:

```
createdAt DateTime    @default(now())
  updatedAt DateTime    @updatedAt
  
  ```
 now generate a second migration, so all previous existing rows now have a default value and new rows will get the correct updatedAt

**Review by dogezen — DISMISSED** [review]

File: prisma/schema.prisma (hunk 3 above)
Comment (Panfi98) [reply prisma/schema.prisma]: fixed
it has two migrations now

Comment (Panfi98) [reply prisma/schema.prisma]: but the first elements breaks the code though

**Review by alinjo — APPROVED** [review]

**Review by JosteinRuen — APPROVED** [review]
Nicely done!

**Review by alihaiderkhannn — APPROVED** [review]
:)

**Pull request merged**